### PR TITLE
feat(external-router): add id to resolvable route

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

There is no way to get the id of an externally resolved entity through the external router driver.

## What is the new behavior?
The type now includes the entity ID, which is useful for downstream drivers.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

`id` is now a required field of `DaffExternallyResolvableUrl`

## Other information

### CODE REVIEW

**Review by damienwebdev — DISMISSED** [review]
lgtm

File: libs/external-router/driver/magento/src/transforms/resolution-to-resolvable-url.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { MagentoUrlResolver } from '../model/url-resolver';
 export const transformResolutionToResolvableUrl = (
   resolution: MagentoUrlResolver,
 ): DaffExternallyResolvableUrl => ({
+  id: resolution.id?.toString() || resolution.entity_uid,
```
Comment (lderrickable) [inline libs/external-router/driver/magento/src/transforms/resolution-to-resolvable-url.ts:8]: I think this means we should now have two different transfomers

Comment (griest024) [reply libs/external-router/driver/magento/src/transforms/resolution-to-resolvable-url.ts:8]: @damienwebdev do you agree?

Comment (damienwebdev) [reply libs/external-router/driver/magento/src/transforms/resolution-to-resolvable-url.ts:8]: Eh, I don't mind splitting the transformer one way or the other, the type is so similar as it's a single key. If this were to expand in later versions, I can foresee a split transformer, but I can't say with any confidence that splitting right now is valuable.

Comment (griest024) [reply libs/external-router/driver/magento/src/transforms/resolution-to-resolvable-url.ts]: well its done now anyway haha

**Review by lderrickable — APPROVED** [review]

**Pull request merged**